### PR TITLE
refactor: consistenly use webdav abstraction layer

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
This makes it easier to remove the SDK in the future.

## Related Issue

- Works towards https://github.com/owncloud/web/issues/9709

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by JammingBen** [comment]
Currently waiting for https://github.com/owncloud/web/pull/9731 and https://github.com/owncloud/web/pull/9721 to be merged first.

File: packages/web-client/src/webdav/getPathForFileId.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { WebDavOptions } from './types'
+
+export const GetPathForFileIdFactory = ({ sdk }: WebDavOptions) => {
+  return {
+    getPathForFileId(id: string): Promise<string> {
+      return sdk.files.getPathForFileId(id)
```
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-client/src/webdav/getPathForFileId.ts:6]: As discussed, this is actually a webdav request.

File: packages/web-client/src/webdav/getPublicFileUrl.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { SpaceResource } from '../helpers'
+import { WebDavOptions } from './types'
+
+export const GetPublicFileUrlFactory = ({ sdk }: WebDavOptions) => {
+  return {
+    getPublicFileUrl(space: SpaceResource, publicLinkToken: string): string {
+      return sdk.publicFiles.getFileUrl(publicLinkToken)
```
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-client/src/webdav/getPublicFileUrl.ts:7]: As discussed, this is actually a webdav request.

File: packages/web-app-files/src/composables/actions/files/useFileActionsEmptyTrashBin.ts [hunk 3]
```diff
@@ -25,12 +24,8 @@ export const useFileActionsEmptyTrashBin = ({ store }: { store?: Store<any> } =
...
 
   const emptyTrashBin = ({ space }: { space: SpaceResource }) => {
-    const path = unref(hasShareJail)
-      ? buildWebDavSpacesTrashPath(space.id)
-      : buildWebDavFilesTrashPath(store.getters.user.id)
-
-    return clientService.owncloudSdk.fileTrash
-      .clearTrashBin(path)
+    return clientService.webdav
+      .clearTrashBin(space, { hasShareJail: unref(hasShareJail), user: store.getters.user })
```
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-app-files/src/composables/actions/files/useFileActionsEmptyTrashBin.ts]: Yes, good idea! I think we need that anyway in the future when we're removing the SDK. Will do in a follow-up.

**Review by dschmidt — APPROVED** [review]

**Pull request merged**